pkg/client/client.go

```diff
@@ -51,71 +51,76 @@ func NewETHClient(endpoint string, opts ...Option) (*ETHClient, error) {
 		o(opt)
 	}
 	return &ETHClient{
-		rpcClient: rpcClient,
-		Client:    ethclient.NewClient(rpcClient),
-		option:    *opt,
+		Client: ethclient.NewClient(rpcClient),
+		option: *opt,
 	}, nil
 }
 
-func (cl *ETHClient) GetTransactionReceipt(ctx context.Context, txHash common.Hash, enableDebugTrace bool) (rc *gethtypes.Receipt, recoverable bool, err error) {
+func (cl *ETHClient) Raw() *rpc.Client {
+	return cl.Client.Client()
+}
+
+func (cl *ETHClient) GetTransactionReceipt(ctx context.Context, txHash common.Hash, enableDebugTrace bool) (rc *gethtypes.Receipt, revertReason string, err error) {
 	var r *Receipt
-	if err := cl.rpcClient.CallContext(ctx, &r, "eth_getTransactionReceipt", txHash); err != nil {
-		return &r.Receipt, true, err
-	}
-	if r == nil {
-		return nil, true, ethereum.NotFound
+
+	if err := cl.Raw().CallContext(ctx, &r, "eth_getTransactionReceipt", txHash); err != nil {
+		return nil, "", err
+	} else if r == nil {
+		return nil, "", ethereum.NotFound
 	} else if r.Status == gethtypes.ReceiptStatusSuccessful {
-		return &r.Receipt, false, nil
+		return &r.Receipt, "", nil
 	} else if r.HasRevertReason() {
 		reason, err := r.GetRevertReason()
-		return &r.Receipt, false, fmt.Errorf("revert: %v(parse-err=%v)", reason, err)
-	} else {
-		errPrefix := "failed to execute a transaction"
-		if enableDebugTrace {
-			to, revertReason, err := cl.DebugTraceTransaction(ctx, txHash)
-			if err != nil {
-				return &r.Receipt, false, fmt.Errorf("%s: %v, txHash: %s, debug_transaction error: %v", errPrefix, r, txHash, err)
-			}
-			return &r.Receipt, false, fmt.Errorf("%s: %v, contract: %s, txHash: %s, revertReason: %v", errPrefix, r, to, txHash, revertReason)
-		} else {
-			return &r.Receipt, false, fmt.Errorf("%s: %v", errPrefix, r)
+		if err != nil {
+			// TODO: use more proper logger
+			logger := log.GetLogger().WithModule("ethereum.chain")
+			logger.Error("failed to get revert reason", err)
+		}
+		return &r.Receipt, reason, nil
+	} else if enableDebugTrace {
+		reason, err := cl.DebugTraceTransaction(ctx, txHash)
+		if err != nil {
+			// TODO: use more proper logger
+			logger := log.GetLogger().WithModule("ethereum.chain")
+			logger.Error("failed to call debug_traceTransaction", err)
 		}
+		return &r.Receipt, reason, nil
+	} else {
+		return &r.Receipt, "", nil
```

Comment:
I think it would be helpful to log that neither the transaction trace nor the reason from the receipt could be obtained.

Reply:
Fixed https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/14/commits/9935faa42fd44b8a81f85c8f13e8ba9779776b4c